app/main.py

```diff
@@ -1,13 +1,27 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [format_linter_error(err) for err in errors],
+            "path": path,
+            "status": "failed" if errors else "passed"
```

Comment:
Same as 13 -15 lines, make your code dry

Reply:
@Oleg-s-s I'm not really clear on what exactly needs to be changed

Reply:
DRY = Dont Repeat Yourself

Reply:
https://blog.devgenius.io/the-dry-principle-in-python-boosting-efficiency-in-machine-learning-c8cf4fded736